File: client/src/api/payments.ts
import { api } from './axios';
import type { Payment, InitiatePaymentInput } from '@/types/payment';
import type { ApiPaginated } from '@/types/api';

export const paymentApi = {
  list: (params: { page?: number; limit?: number; saleId?: string; status?: string } = {}) =>
    api
      .get<ApiPaginated<Payment>>('/client/payments', { params })
      .then((r) => r.data),

  initiate: (payload: InitiatePaymentInput) =>
    api
      .post<{ data: { paymentId: string; checkoutRequestId?: string; message?: string } }>(
        '/client/payments/initiate',
        payload
      )
      .then((r) => r.data.data),

  recordManual: (payload: {
    saleId: string;
    method: string;
    amount?: number;
    reference?: string;
    note?: string;
    amountReceived?: number;
  }) =>
    api.post<{ data: Payment }>('/client/payments/manual', payload).then((r) => r.data.data),

  refund: (id: string, reason?: string) =>
    api
      .post<{ data: Payment }>(`/client/payments/${id}/refund`, { reason })
      .then((r) => r.data.data),
};

File: Android/client/lib/api/payments.dart
import 'api_client.dart';

class PaymentsApi {
  static final _api = BizOsApiClient.instance;

  static Future<Response<dynamic>> list({Map<String, dynamic>? query}) =>
      _api.get('/client/payments', query: query);
  static Future<Response<dynamic>> initiate(Map<String, dynamic> data) =>
      _api.post('/client/payments/initiate', data: data);
  static Future<Response<dynamic>> manual(Map<String, dynamic> data) =>
      _api.post('/client/payments/manual', data: data);
  static Future<Response<dynamic>> refund(
    String id, {
    Map<String, dynamic>? data,
  }) => _api.post('/client/payments/$id/refund', data: data);
}
